plugins/outputs/cloudwatch/cloudwatch_test.go

```diff
@@ -323,45 +323,43 @@ func (svc *mockCloudWatchClient) PutMetricData(input *cloudwatch.PutMetricDataIn
 	return args.Get(0).(*cloudwatch.PutMetricDataOutput), args.Error(1)
 }
 
-func newCloudWatchClient(svc cloudwatchiface.CloudWatchAPI) *CloudWatch {
+func newCloudWatchClient(svc cloudwatchiface.CloudWatchAPI, forceFlushInterval time.Duration) *CloudWatch {
 	cloudwatch := &CloudWatch{
 		svc:                svc,
-		ForceFlushInterval: internal.Duration{Duration: 1 * time.Second},
+		ForceFlushInterval: internal.Duration{Duration: forceFlushInterval},
 	}
 	cloudwatch.startRoutines()
 	return cloudwatch
 }
 
-func TestWrite(t *testing.T) {
```

Comment:
In case this diff is confusing, what I did was remove some common code from `TestWrite()` and `TestWriteError()` and put it in a new `makeMetrics()` function. And I use the new function in the new test case `TestPublish()` as well.